core-rust/core-api-server/src/core_api/conversions/lts.rs

```diff
@@ -365,13 +371,99 @@ pub fn to_api_lts_fungible_resource_balance_change(
 }
 
 pub fn to_api_lts_resultant_account_fungible_balances(
-    _database: &StateManagerDatabase,
-    _context: &MappingContext,
-    _substate_changes: &BySubstate<ChangeAction>,
+    database: &StateManagerDatabase,
+    context: &MappingContext,
+    substate_changes: &BySubstate<ChangeAction>,
 ) -> Vec<models::LtsResultantAccountFungibleBalances> {
-    // TODO(parallel PR): go through substate changes, collect fungible vaults' new values, resolve
-    // each into its global component owning the vault, retain only accounts, adjust API docs.
-    vec![]
+    let fungible_vaults = substate_changes
+        .iter_node_ids()
+        .filter(|node_id| node_id.is_internal_fungible_vault())
+        .collect::<Vec<_>>();
+    let ancestries = database
+        .batch_get_ancestry(fungible_vaults.clone())
+        .into_iter()
+        .map(|record| record.expect("InternalFungibleVault does not have a parent"));
+    let node_id_to_ancestor: HashMap<_, _> = fungible_vaults.into_iter().zip(ancestries).collect();
+    let system_db_reader = SystemDatabaseReader::new(database);
+    let mut resultant_fungible_balances_by_account = HashMap::new();
+    substate_changes
+        .iter()
+        .filter_map(|(substate_reference, change_action)| {
```

Comment:
can we elegantly extract at least some large part from this closure? (into some `calculate_resultant_balance()`)

Reply:
I'm not sure I can extract anything (reusable/useful), however I've refactored it to use multiple `filter`, `filter_map` and `map`s which are clearly showing the progression from `(substate_reference, change_action)` into `(account_address, resource_address, resultant_balance)`. 

I hope this is an improvement.